android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/CustomPhotoCaptureFactory.kt

```diff
@@ -0,0 +1,136 @@
+import kotlinx.coroutines.launch
+import org.hl7.fhir.r4.model.QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent
+import org.hl7.fhir.r4.model.StringType
+import org.smartregister.fhircore.engine.R
+import org.smartregister.fhircore.engine.util.extension.encodeToBase64
+import org.smartregister.fhircore.engine.util.extension.hide
+import org.smartregister.fhircore.engine.util.extension.show
+
+class CustomPhotoCaptureFactory(fragment: Fragment) :
+  QuestionnaireItemViewHolderFactory(R.layout.custom_photo_capture_layout) {
+
+  private lateinit var tvPrefix: TextView
+  private lateinit var tvHeader: TextView
+  private lateinit var ivThumbnail: ImageView
+  private lateinit var btnTakePhoto: MaterialButton
+  private lateinit var launchCamera: ActivityResultLauncher<Intent>
+  var questionnaireResponse = QuestionnaireResponseItemAnswerComponent()
+
+  init {
+    registerCameraIntent(fragment)
+  }
+
+  private fun registerCameraIntent(fragment: Fragment) {
+    with(fragment) {
+      launchCamera =
+        registerForActivityResult(StartActivityForResult()) { result ->
+          if (result.resultCode == Activity.RESULT_OK) {
+            val imageBitmap = result.data?.extras?.get(EXTRA_IMAGE) as Bitmap
+            ivThumbnail.apply {
+              Glide.with(fragment).load(imageBitmap).into(this)
+              show()
+            }
+            btnTakePhoto.text = getString(R.string.replace_photo)
+            lifecycleScope.launch(Dispatchers.Main) {
+              val imageFile =
+                async(Dispatchers.IO) {
+                  val tempFile =
+                    File.createTempFile(PREFIX_BITMAP, ".${Bitmap.CompressFormat.JPEG.extension()}")
+                  tempFile.deleteOnExit()
+                  saveBitmap(imageBitmap, tempFile)
+                  return@async tempFile
+                }
+              val imageFileCompressed =
+                compress(requireContext(), imageFile.await()) { size(MAX_COMPRESSION_SIZE) }
+              val imageBase64 = imageFileCompressed.encodeToBase64()
+              questionnaireResponse.value = StringType(imageBase64)
+            }
+          }
+        }
+    }
+  }
+
+  override fun getQuestionnaireItemViewHolderDelegate(): QuestionnaireItemViewHolderDelegate =
+    object : QuestionnaireItemViewHolderDelegate {
+
+      override lateinit var questionnaireItemViewItem: QuestionnaireItemViewItem
+
+      override fun init(itemView: View) {
+        tvPrefix = itemView.findViewById(R.id.tv_prefix)
+        tvHeader = itemView.findViewById(R.id.tv_header)
+        ivThumbnail = itemView.findViewById(R.id.iv_thumbnail)
+        btnTakePhoto = itemView.findViewById(R.id.btn_take_photo)
+      }
+
+      override fun bind(questionnaireItemViewItem: QuestionnaireItemViewItem) {
+        if (!questionnaireItemViewItem.questionnaireItem.prefix.isNullOrEmpty()) {
+          tvPrefix.show()
+          tvPrefix.text = questionnaireItemViewItem.questionnaireItem.prefix
```

Comment:
Scope function `apply` would make this code more readable

Reply:
Okay